### ISSUE TITLE
Bladeburner/Sleeve - unequal naming GUI/scripts

### ISSUE THREAD

**Issue by NoCanDouCan** [bitburner-official/bitburner-src#372]
There are some minor naming issues with the Bladeburner/Sleeve GUI and the script commands.
Some examples:

1. Bladeburner GUI
There are 5 major categories of tasks (General, **_Contracts_**, **_Operations_**, BlackOps,Skills)
If we use the function ns.bladeburner.getCurrentAction() the field type will be filled with
(General, **_Contract_**, **_Operation_**,BlackOps).

2. Sleeve GUI 
The task for the sleeve is called "Field analysis". In the Bladeburner GUI its called "Field Analysis".
And the script command would be ns.sleeve.setToBladeburnerAction(0,"Field analysis")

3. Sleeve script
ns.sleeve.getTask(i); will show "Contracts" and ns.sleeve.setToBladeburnerAction(i,"Take on contracts","Bounty Hunter");

I would recommend to use the same naming for all GUI and script commands .

- Version of the game: Bitburner v2.2.1 Steam

**Comment by Snarling** [comment]
Totally agree. Unfortunately for things like this it will cause an API break if we standardize the internal code to match the UI names - ran into that with some similar changes in Corporation and Crime a couple patches back. Further renaming of stuff that affects API usage will be put off until 2.3

**Comment by Zelow79** [comment]
For cases 2 and 3 the functions do actually match the GUI. The issue is you're considering the BB UI and not the Sleeve UI those come from.

![image](https://user-images.githubusercontent.com/32428876/220454750-8bc3da44-f26a-4107-a710-887080ed206f.png)
![image](https://user-images.githubusercontent.com/32428876/220454788-cb279924-9c30-45d0-bacd-ebff611df190.png)

**Comment by d0sboots** [comment]
Perhaps the BB and Sleeve UIs should be standardized against each other?

**Comment by d0sboots** [comment]
Also, it's *possible* to do this without an API break: You can continue to accept the old values as synonyms for the new values. Whether the compatibility is worth the hygiene hit is a judgement call.

**Comment by Snarling** [comment]
If there is a name for a task or category of tasks in bladeburner that is also present for sleeves, it should definitely use consistent naming in both places.

On the topic of permissive processing of player input, see how player input for Crimes are currently handled (probably way too permissive):

```js
export function findCrime(roughName: string): Crime | null {
  if (checkEnum(CrimeType, roughName)) return Crimes[roughName];
  roughName = roughName.toLowerCase();
  if (roughName.includes("shoplift")) return Crimes[CrimeType.shoplift];
  else if (roughName.includes("rob") && roughName.includes("store")) return Crimes[CrimeType.robStore];
  else if (roughName.includes("mug")) return Crimes[CrimeType.mug];
  else if (roughName.includes("larceny")) return Crimes[CrimeType.larceny];
  else if (roughName.includes("drugs")) return Crimes[CrimeType.dealDrugs];
  else if (roughName.includes("bond") && roughName.includes("forge")) return Crimes[CrimeType.bondForgery];
  else if ((roughName.includes("traffic") || roughName.includes("illegal")) && roughName.includes("arms")) {
    return Crimes[CrimeType.traffickArms];
  } else if (roughName.includes("homicide")) return Crimes[CrimeType.homicide];
  else if (roughName.includes("grand") && roughName.includes("auto")) return Crimes[CrimeType.grandTheftAuto];
  else if (roughName.includes("kidnap")) return Crimes[CrimeType.kidnap];
  else if (roughName.includes("assassin")) return Crimes[CrimeType.assassination];
  else if (roughName.includes("heist")) return Crimes[CrimeType.heist];
  return null;
}
```

Permissive acceptance like this is a double edged sword. One one hand it makes the players scripts work as long as they are close enough, so that's good. And it allows slight changes in the enum value without it causing an API break.

But then it enforces players not caring about exactness. In also causes a slight slowdown on using these API functions since you have to process the player's input through an additional function. If 2.3 is going to include sufficient ways for the player to access the "real names" of stuff like e.g. crimes, I think this permissive code can also be removed and we can expect users to actually use the exact names we are exposing to them.

Another alternative is to always convert player input for these things to lowercase with no spaces, and have a helper that maps those specific inputs to the actual enum members. Way faster than this if/elseif/elseif/elseif hardcoding above, while still maintaining some level of permissiveness.

**Comment by Zelow79** [comment]
> If there is a name for a task or category of tasks in bladeburner that is also present for sleeves, it should definitely use consistent naming in both places.

Aside from Field Analysis here, it isn't really the task. It's the type "Contracts" vs "Take on contracts". Which makes complete sense for a sleeve doing BB Contracts. The tasks Bounty Hunter, Investigation,  and Retirement should be correct.

**Comment by Snarling** [comment]
So is this the usage of each?
```js
  ns.bladeburner.startAction("Contracts", "Tracking");
  ns.sleeve.setToBladeburnerAction(0, "Take on Contracts", "Tracking");
```
Seems like a discrepancy to me. Shouldn't it be the same format either way? In either case the person being told to perform an action will be "taking on" that action

**Comment by Zelow79** [comment]
The difference is who is doing it, do you want that sleeve drop down window to say "Perform Bladeburner actions" - "Contracts" - "Investigation"? I guess it comes down to personal opinion here really, but I think the later sounds sillier for the sleeves perspective.

**Comment by cmsautter** [comment]
> There are some minor naming issues with the Bladeburner/Sleeve GUI and the script commands. Some examples:
> 
>     1. Bladeburner GUI
>        There are 5 major categories of tasks (General, **_Contracts_**, **_Operations_**, BlackOps,Skills)
>        If we use the function ns.bladeburner.getCurrentAction() the field type will be filled with
>        (General, **_Contract_**, **_Operation_**,**BlackOps**).

No, ns.bladeburner.getCurrentAction() returns "Black Operations" not "BlackOps", atleast in v2.6.2
This is another inconsistency that should be fixed at some time.
_(Note: ns.bladeburner.startAction("BlackOps", ..) works though)_
